compiler/next/include/chpl/uast/ShadowVariable.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_SHADOWVARIABLE_H
+#define CHPL_UAST_SHADOWVARIABLE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Formal.h"
+#include "chpl/uast/VarLikeDecl.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a shadow variable. Shadow variables are declared
+  in with clauses. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      var a: atomic int;
+      forall x in 1..15 with (ref a) do
+        a.fetchAdd(x);
+  \endrst
+
+  Creates a shadow variable 'a' which refers to the outer variable 'a' by
+  ref intent.
+ */
+class ShadowVariable final : public VarLikeDecl {
+ public:
+  enum Intent {
+    VAR,
+    CONST,
+    CONST_REF,
+    REF,
+    IN = Formal::IN,
+    CONST_IN = Formal::CONST_IN,
```

Comment:
Since the difference seems to be mainly the presence of a type/init expression, wouldn't it make more sense to perhaps rename `ShadowVar` to `TaskVar` and then resolve it to either a shadow variable or a task private variable at a later point in time?